test/integration/udproute_test.go

```diff
@@ -75,3 +90,62 @@ func TestUDPRouteBasics(t *testing.T) {
 	require.NoError(t, err)
 	require.Contains(t, output.String(), message)
 }
+
+func TestUDPRouteNoReach(t *testing.T) {
+	udpRouteNoReachCleanupKey := "udproutenoreach"
+	defer func() {
+		utils.DumpDiagnosticsIfFailed(ctx, t , env.Cluster())
+		runCleanup(udpRouteNoReachCleanupKey)
+	}()
+
+	t.Log("deploying config/samples/udproute-noreach kustomize")
+	require.NoError(t, clusters.KustomizeDeployForCluster(ctx, env.Cluster(), udproutenoreachSampleKustomize))
+	addCleanup(udpRouteNoReachCleanupKey, func(ctx context.Context) error {
+		cleanupLog("cleaning up config/samples/udproute-noreach kustomize")
+		return clusters.KustomizeDeleteForCluster(ctx, env.Cluster(), udproutenoreachSampleKustomize)
+	})
+
+	// TODO (astoycos) this currently won't work but it will with updated control-plane logic
+	// add it back so we don't have to maintain a whole new kustomize config for a one line change.
+	// // Update Server to ensure it's in dry run mode then wait for it to be ready
+	// server, err := env.Cluster().Client().AppsV1().Deployments(corev1.NamespaceDefault).Get(ctx, udprouteSampleName, metav1.GetOptions{})
+	// require.NoError(t, err)
+	// server.Spec.Template.Spec.Containers[0].Command = []string{"./udp-test-server", "--dry-run"}
+	// _, err = env.Cluster().Client().AppsV1().Deployments(corev1.NamespaceDefault).Update(ctx, server, metav1.UpdateOptions{})
+	// require.NoError(t, err)
```

Comment:
This TODO would appear to warrant a follow-up issue.

Reply:
This TODO isn't super relevant anymore I like how the Kustomize config turned out with using patches.